data_structures/binary_tree_height.py

```diff
@@ -0,0 +1,172 @@
+# Find the binary tree height
+
+
+import random
+
+
+# Node of the Binary Tree
+
+
+class Node:
+    def __init__(self, data) -> None:
+        self.data = data
+        self.left = None
+        self.right = None
+        self.height = 0
+
+    def __str__(self):
+        lines, *_ = self._display_aux()
+        return "\n".join(lines)
+
+    def _display_aux(self):
+        # No child.
+        if self.right is None and self.left is None:
+            if self.height:
+                line = "({}, {})".format(self.data, self.height)
+            else:
+                line = "({})".format(self.data)
+            width = len(line)
+            height = 1
+            middle = width // 2
+            return [line], width, height, middle
+
+        # Only left child.
+        if self.right is None:
+            lines, n, p, x = self.left._display_aux()
+            if self.height > 0:
+                s = "({}, {})".format(self.data, self.height)
+            else:
+                s = "({})".format(self.data)
+            u = len(s)
+            first_line = (x + 1) * " " + (n - x - 1) * "-" + s
+            second_line = x * " " + "/" + (n - x - 1 + u) * " "
+            shifted_lines = [line + u * " " for line in lines]
+            return [first_line, second_line] + shifted_lines, n + u, p + 2, n + u // 2
+
+        # Only right child.
+        if self.left is None:
+            lines, n, p, x = self.right._display_aux()
+            if self.height:
+                s = "({}, {})".format(self.data, self.height)
+            else:
+                s = "({})".format(self.data)
+            u = len(s)
+            first_line = s + x * "-" + (n - x) * " "
+            second_line = (u + x) * " " + "\\" + (n - x - 1) * " "
+            shifted_lines = [u * " " + line for line in lines]
+            return [first_line, second_line] + shifted_lines, n + u, p + 2, u // 2
+
+        # Two children.
+        left, n, p, x = self.left._display_aux()
+        right, m, q, y = self.right._display_aux()
+        if self.height:
+            s = "({}, {})".format(self.data, self.height)
+        else:
+            s = "({})".format(self.data)
+        u = len(s)
+        first_line = (x + 1) * " " + (n - x - 1) * "-" + s + y * "-" + (m - y) * " "
+        second_line = (
+            x * " " + "/" + (n - x - 1 + u + y) * " " + "\\" + (m - y - 1) * " "
+        )
+        if p < q:
+            left += [n * " "] * (q - p)
+        elif q < p:
+            right += [m * " "] * (p - q)
+        zipped_lines = zip(left, right)
+        lines = [first_line, second_line] + [a + u * " " + b for a, b in zipped_lines]
+        return lines, n + m + u, max(p, q) + 2, n + u // 2
+
+
+# Binary Tree
+class BinaryTree:
```

Comment:
Stop reimplementing this. I am not going to review it.

You can reimplement it for learning but remove and replace with an imported one that is central.